dashboard/app/controllers/potential_teachers_controller.rb

```diff
@@ -28,6 +31,15 @@ def show
     params.permit([:name, :email, :script_id, :receives_marketing]).to_h
   end
 
+  def send_hoc_email(params)
+    name = current_user&.email || params[:email]
+    email = current_user&.name || params[:name]
+    unit_id = params[:script_id]
+    lessons = Unit.find_by_id(unit_id).lessons
```

Comment:
nit: using `Unit.get_from_cache` to take advantage of our caching

Reply:
Updated- thanks!